README.md

```diff
@@ -123,51 +109,64 @@ myDb.on('ready', function () {
 })
 ```
 
-#### `unwatch = db.watch(folderOrKey, onchange)`
+#### `watcher = db.watch(folderOrKey, onchange)`
 
 Watch a folder and get notified anytime a key inside this folder
 has changed.
 
-``` js
-db.watch('/foo/bar', function () {
+```js
+db.watch('foo/bar', function () {
   console.log('folder has changed')
 })
 
 ...
 
-db.put('/foo/bar/baz', 'hi') // triggers the above
+db.put('foo/bar/baz', 'hi') // triggers the above
 ```
 
-#### `var stream = db.createDiffStream(key[, checkout][, head])`
+You can destroy the watcher by calling `watcher.destroy()`.
 
-Find out about changes in key/value pairs between the version `checkout` and
-`head` for all keys prefixed by `key`.
+The watcher will emit `watching` when it starts watching and `change`
+when a change has been detected.
 
-`checkout` and `head` are versions to use to compare against. If not provided,
-`head` is the current HEAD of the database, and `checkout` is the beginning of
-time.
+If a critical error occurs an error will be emitted on the watcher.
 
-`stream` is a readable object stream that outputs modifications like
+#### `var stream = db.createReadStream(prefix[, options])`
+
+Create a readable stream of nodes stored in the database.
+Set `prefix` to only iterate nodes prefixed with that folder.
+
+Options include:
 
 ```js
-{ type: 'del', name: '/a', nodes: [<node>] },
-{ type: 'put', name: '/a', nodes: [<node>, <node>] }
-{ type: 'put', name: '/b/beep', value: [<node>] }
+{
+  recursive: true // visit all subfolders.
+                  // set to false to only visit the first node in each folder
+}
 ```
 
-that occured between `checkout` and `head`. When multiple feeds conflict for the
-value of a key at a point in time, `nodes` will have multiple entries. `<node>`
-is the full hyperdb node.
+#### `db.list(prefix[, options], callback)`
+
+Same as `createReadStream` but buffers the result to a list that is passed to the
+callback.
+
+#### `var stream = db.createDiffStream(prefix[, checkout)`
+
+Find out about changes in key/value pairs between the version `checkout` and
+current version prefixed by `prefix`.
 
-#### `var stream = db.createHistoryStream([opts, ][start])`
+`stream` is a readable object stream that outputs modifications like
+
+```js
+{ left: nodes, right: nodes }
+```
 
-Returns a readable stream of node objects covering all historic values since
-either the beginning of time, or the [version](#dbversioncallback) `opts.start`.
+`left` are the nodes for a key found in the `db` and `right` are the nodes found in the `checkout`.
+If no nodes exist in the `db` for the key `left` will be `null` and vice versa.
 
-Valid opts include:
+#### `var stream = db.createHistoryStream()`
```

Comment:
Appears here in the README but is not exposed by the API.

What's the plan re feature parity? Develop on master until there's no more major semver breakage? I suggest keeping the readme api consistent with the implementation as you go, so it stays clear what's implemented and what isn't.